282 thumbnail selector

billbitt opened the pull request:
- for publishing videos: added a slider to select a thumbnail from the video
- moved the publish functionality out of the publish component and into a Redux saga
- added check on the back end publish route to check for and publish the thumbnail file if present

seanyesmunt commented on react/actions/publish.js:
```diff
@@ -71,3 +71,18 @@ export function toggleMetadataInputs (showMetadataInputs) {
 };
+
+export function onNewThumbnail (file) {
+  console.log('new thumbnail action created');
```
Not needed

seanyesmunt commented on react/channels/publish.js:
```diff
@@ -0,0 +1,52 @@
+import {buffers, END, eventChannel} from 'redux-saga';
+
+export const makePublishRequestChannel = (fd) => {
+  console.log('making publish request');
```
Not needed

seanyesmunt commented on routes/api-routes.js:
```diff
@@ -135,14 +135,14 @@ module.exports = (app) => {
     logger.debug('api/claim-publish files:', files);
     // define variables
-    let  name, fileName, filePath, fileType, nsfw, license, title, description, thumbnail, channelName, channelPassword;
+    let  name, fileName, filePath, fileType, thumbnailFileName, thumbnailFilePath, thumbnailFileType, nsfw, license, title, description, thumbnail, channelName, channelPassword;
```
Why does spee.ch need to know the `thumbnailFilePath` on the server?

billbitt replied:
This is because I'm having the client send the the thumbnail file along with the main video file together from the front end and creating the thumbnail data dynamically (if needed) based off of the main video file's metadata on the back end.

billbitt commented:
- removed the unneeded `browserHistory.js` file 
- cleaned up the console logs
- the vendor files were left over from when I was using masonry.js; I may add this back but they are not being used currently so I removed them
